mbuild/utils/conversion.py

```diff
@@ -2,26 +2,98 @@
 import numpy as np
 
 
-def RB_to_OPLS(c0, c1, c2, c3, c4, c5):
-    """Convert Ryckaert-Bellemans type dihedrals to OPLS type.
+def RB_to_OPLS(c0, c1, c2, c3, c4, c5, error_tol=1e-4):
+    r"""Convert Ryckaert-Bellemans type dihedrals to OPLS type.
+
+    .. math::
+    RB_torsions &= c0 + c1*cos(psi) + c2*cos(psi)^2 + c3*cos(psi)^3 + \\
+                &= c4*cos(psi)^4 + c5*cos(5*psi)^5
+
+    .. math::
+    OPLS_torsions &= f1/2*(1+cos(t)) + f2/2*(1-cos(2*t)) + \\
+                  &= f3/2*(1+cos(3*t)) + f4/2(1-cos(4*t))
+
+    where :math:`psi = t - pi = t - 180 degrees`
 
     Parameters
     ----------
     c0, c1, c2, c3, c4, c5 : Ryckaert-Belleman coefficients (in kcal/mol)
+    error_tol : float, default=1e-4
+        The acceptable absolute tolerance between the RB to OPLS conversion.
+        Any value entered is converted to an absolute value.
 
     Returns
     -------
     opls_coeffs : np.array, shape=(4,)
         Array containing the OPLS dihedrals coeffs f1, f2, f3, and f4
-        (in kcal/mol)
+        (in kcal/mol).
+        NOTE: fO IS NOT IN THE OPLS DIHEDRAL EQUATION AND IS ONLY USED
+        TO TEST IF THIS FUNCTION CAN BE UTILIZED. ONE OF THE
+        CONDITIONS OF CONVERSION IS f0 MUST BE ZERO (f0=0).
+
+    Notes
+    -----
+    c5 must equal zero, or this conversion is not possible.
+
+    (c0 + c1 + c2 + c3 + c4 + c5) must equal zero, or this conversion is not possible.
     """
-    f1 = (-1.5 * c3) - (2 * c1)
-    f2 = c0 + c1 + c3
-    f3 = -0.5 * c3
-    f4 = -0.25 * c4
+    if isinstance(error_tol, float):
+        error_tol = abs(error_tol)
+    else:
+        raise TypeError("ERROR: The error_tol variable must be a float.")
+
+    if bool(np.isclose(c5, 0, atol=error_tol, rtol=0)) is False:
```

Comment:
We can use [the numpy.all method to perform the same thing](https://numpy.org/doc/stable/reference/generated/numpy.all.html#numpy.all). Without an axis defined, the behavior is the same. and follows most of our other test setups.
```suggestion
    if not np.all(np.isclose(c5, 0, atol=error_tol, rtol=0)):
```

All instances of a numpy array simplifying down to a single boolean, is a bit easier to read using the numpy.all command imo.

Reply:
changed